packages/devtools_app/lib/src/screens/app_size/app_size_controller.dart

```diff
@@ -238,22 +270,64 @@ class AppSizeController {
     changeAnalysisJsonFile(jsonFile);
 
     // Set deferred app flag.
-    isDeferredApp =
-        deferredLoadingSupportEnabled && processedJson['n'] == 'ArtificialRoot';
+    _isDeferredApp.value =
+        deferredLoadingSupportEnabled && _hasDeferredInfo(processedJson);
+
+    if (isDeferredApp.value) {
+      _deferredOnly = _extractDeferredUnits({...processedJson});
```

Comment:
can we just write `processedJson` instead of `{...processedJson}`?

Reply:
I'm using `{...processedJson}` to make a deep-copy of the `processedJson`, so that we aren't modifying the underlying data

Reply:
You can also use `Map.from` which might make the intent more clear

Reply:
Done!